packages/gatsby/src/utils/api-browser-docs.js

```diff
@@ -35,6 +35,11 @@ exports.onRouteUpdate = true
  * @param {object} $0
  * @param {object} $0.prevRouterProps The previous state of the router before the route change.
  * @param {object} $0.pathname The new pathname
+ * @example
+ * exports.shouldUpdateScroll = (prevRouterProps, pathname) => {
+ *   window.scrollTo(0, 0)  // upon navigation, scroll to top of page
```

Comment:
I'm curious what the use case for doing this would be?

Reply:
For my particular use-case, I wanted it so when navlinks clicked always start at the top of the page.  By default, they start off at the last scrolled position of the page.  This example overrides that behavior.

Reply:
Could you try upgrading to the latest Gatsby? There was a bug that was causing problems with scrolling that I fixed in https://github.com/gatsbyjs/gatsby/pull/3775

It sounds like you were just running into that bug.

I'd love an example for this API but it should be a realistic one — not one that works around a bug :-)

Reply:
Understood!  I'll remove my workaround, and then test.  If I come up with a better example, I will absolutely submit.  Thanks for the update, Kyle.

Reply:
Just confirming you fixed the issue I was having with your patch.